src/test/java/io/neonbee/internal/json/ConfigurableJsonFactoryClusterTest.java

```diff
@@ -0,0 +1,122 @@
+    }
+
+    @AfterAll
+    static void resetToDefault() {
+        ConfigurableJsonFactory.CODEC.setMaxStringLength(DEFAULT_MAX_STRING_LEN);
+    }
+
+    @Test
+    @Timeout(value = 5, timeUnit = TimeUnit.SECONDS)
+    @DisplayName("Sending a JSON that on serialization will result in a JSON that exceeds the maxStringLength should raise a descriptive error message")
+    void testSendDescriptiveJsonError(
+            @NeonBeeInstanceConfiguration(clustered = true, activeProfiles = {}) NeonBee nb1,
+            @NeonBeeInstanceConfiguration(clustered = true, activeProfiles = {}) NeonBee nb2,
+            VertxTestContext testContext) {
+        DataVerticle<JsonObject> testVerticle = new DataVerticle<>() {
+            @Override
+            public String getName() {
+                return "TestVerticle";
+            }
+
+            @Override
+            public Future<JsonObject> retrieveData(DataQuery query, DataContext context) {
+                return Future.succeededFuture();
+            }
+        };
+
+        nb2.getVertx().deployVerticle(testVerticle).map(any -> new DataRequest(testVerticle.getName(),
+                new DataQuery().setBody(createJsonObjectWithLengthOver(SMALLER_JSON_LENGTH, false).toBuffer())))
+                .compose(dr -> DataVerticle.requestData(nb1.getVertx(), dr, new DataContextImpl()))
+                .onComplete(testContext.failing(throwable -> {
+                    assertStringLengthExceptionMessage(throwable);
+                    testContext.completeNow();
+                }));
+    }
+
+    @Test
+    @Timeout(value = 5, timeUnit = TimeUnit.SECONDS)
+    @DisplayName("Using a JSON in the target verticle that exceeds the maxStringLength should raise a descriptive error message")
+    void testRetrieveDescriptiveJsonError(
+            @NeonBeeInstanceConfiguration(clustered = true, activeProfiles = {}) NeonBee nb1,
+            @NeonBeeInstanceConfiguration(clustered = true, activeProfiles = {}) NeonBee nb2,
+            VertxTestContext testContext) {
+        DataVerticle<JsonObject> testVerticle = new DataVerticle<>() {
+            @Override
+            public String getName() {
+                return "TestVerticle";
+            }
+
+            @Override
+            public Future<JsonObject> retrieveData(DataQuery query, DataContext context) {
+                return Future.succeededFuture(createJsonObjectWithLengthOver(SMALLER_JSON_LENGTH, true));
+            }
+        };
+
+        nb2.getVertx().deployVerticle(testVerticle).map(any -> new DataRequest(testVerticle.getName()))
+                .compose(dr -> DataVerticle.requestData(nb1.getVertx(), dr, new DataContextImpl()))
+                .onComplete(testContext.failing(throwable -> {
+                    assertStringLengthExceptionMessage(throwable);
+                    testContext.completeNow();
+                }));
+    }
+
+    @Test
+    // @Timeout(value = 5, timeUnit = TimeUnit.SECONDS)
```

Comment:
why is the timeout commented? guess we should comment it in again.

Reply:
I think we need to remove all timeouts, the timeout is controlled by the environment, since https://github.com/SAP/neonbee/commit/02d22528